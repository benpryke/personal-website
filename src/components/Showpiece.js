import React from 'react';

/**
 * A piece to show off, with an image, a title and some text
 */
export default class Showpiece extends React.Component {
  render() {
    const { title, thumbnail, description, url } = this.props;

    return (
      <a href={url}>
        <div className='showpiece'>
          <img src={thumbnail} alt={title}/>
          <h3>{title}</h3>
          <p>{description}</p>
        </div>
      </a>
    );
  }
}
