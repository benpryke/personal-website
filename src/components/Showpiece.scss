.showpiece {
  position: relative;
  top: 0;
  margin: 20px;
  width: 300px;
  height: 400px;
  padding: 10px;
  background: #fff;
  border-radius: 10px;
  box-shadow: 8px 14px 38px rgba(45, 51, 56, 0.08), 1px 3px 8px rgba(45, 51, 56, 0.03);
  transition: 0.5s ease;
  
  @media (hover: hover) {
    &:hover {
      top: -10px;
    }
  }

  p {
    color: $primary-text-colour;
  }

  img {
    width: 300px;
    height: 200px;
    object-fit: cover;
  }
}
