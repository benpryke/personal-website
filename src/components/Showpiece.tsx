import React from "react";
import LazyLoad from "react-lazyload";

export interface ShowpieceProps {
  /** Heading to display */
  title: string;
  /** URL for the thumbnail image */
  thumbnail: string;
  /** Body text of the showpiece */
  description: string;
  /** Clicking on the showpiece opens this URL */
  url: string;
}

/**
 * A piece to show off, with an image, a title and some text
 */
export const Showpiece: React.FC<ShowpieceProps> = ({
  title,
  thumbnail,
  description,
  url,
}) => {
  return (
    <a href={url}>
      <div className="showpiece">
        <LazyLoad height={200} offset={100}>
          <img src={thumbnail} alt={title} />
        </LazyLoad>
        <h3>{title}</h3>
        <p>{description}</p>
      </div>
    </a>
  );
};
