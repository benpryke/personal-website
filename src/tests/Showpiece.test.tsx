import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import Showpiece, { ShowpieceProps } from "../components/Showpiece";

const props: ShowpieceProps = {
  title: "title",
  thumbnail: "thumbnail",
  description: "description",
  url: "url",
};

describe("Showpiece", () => {
  it("renders without crashing", () => {
    render(<Showpiece {...props} />);
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<Showpiece {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
