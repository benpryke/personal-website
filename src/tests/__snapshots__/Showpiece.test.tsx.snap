// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showpiece should match the snapshot 1`] = `
<a
  href="url"
>
  <div
    className="showpiece"
  >
    <div
      className="lazyload-placeholder"
      style={
        Object {
          "height": 200,
        }
      }
    />
    <h3>
      title
    </h3>
    <p>
      description
    </p>
  </div>
</a>
`;
